Library/Homebrew/macos_version.rb

```diff
@@ -66,6 +66,19 @@ def <=>(other)
     result
   end
 
+  sig { params(comparator: String, other: T.any(Version, T::Array[MacOSVersion])).returns(T::Boolean) }
+  def compare(comparator, other)
```

Comment:
I'd be okay with calling this something else (and replacing the call to `super` appropriately) since the semantics are a bit looser than comparison. `compare_or_included` or something. Though, since the method becomes a bit more specific, it could go into `cask.rb` instead of `macos_version.rb`.